rclpy/test/test_destruction.py

```diff
@@ -261,3 +261,56 @@ def test_destroy_timer_asap():
             node.destroy_node()
     finally:
         rclpy.shutdown(context=context)
+
+
+triggered_function = False
+
+
+def test_context_on_shutdown_function():
+
+    class TestShutdown():
+
+        def __init__(self):
+            self.callback_triggered = False
+
+        def on_shutdown_method(self):
+            self.callback_triggered = True
+
+    def callback_function():
+        global triggered_function
+        triggered_function = True
```

Comment:
Is the repetitive `triggered_function` declaration intentional?

Reply:
The global declaration needs to be done before you use it in the scope to tell the interpreter how to react to the reference.